### PR TITLE
Add tests to internal/render/sse/stream

### PR DESCRIPTION
**Pull request by lechengfan**
- Add tests to test all methods of Stream, which were missing previously.

### CODE REVIEW

File: services/horizon/internal/render/sse/stream_test.go [hunk 1]
```diff
@@ -1,24 +1,126 @@
...
+
+// Tests that the stream sends the preamble before any events and that events are correctly sent.
+func (suite *StreamTestSuite) TestStream_Send() {
+	e := Event{Data: "test message"}
+	suite.stream.Send(e)
+	suite.stream.Done()
+	// Before sending, it should have sent the preamble first and set the headers.
+	suite.checkHeadersAndPreamble()
+	// Now check that the data got written
+	assert.Contains(suite.T(), suite.w.Body.String(), "test message")
```
Comment (bartekn) [inline services/horizon/internal/render/sse/stream_test.go]: We should check the full event string so: `data: test message\n\n` and probably before `suite.stream.Done()`.

File: services/horizon/internal/render/sse/stream_test.go [hunk 2]
```diff
@@ -1,24 +1,126 @@
...
-	stream.interval = 500 * time.Millisecond
-	stream.Init()
+	suite.stream.(*stream).interval = 500 * time.Millisecond
+	suite.stream.Init()
 	// Wait long enough for heartbeat to send
 	time.Sleep(time.Second)
-	stream.Done()
-	assert.Contains(t, w.Body.String(), "\n:heartbeat")
+	suite.stream.Done()
+	assert.Contains(suite.T(), suite.w.Body.String(), "\n:heartbeat")
```
Comment (bartekn) [inline services/horizon/internal/render/sse/stream_test.go]: Good idea to check if `httptest.ResponseRecorder.Flushed` is set to try to ensure that `Flush()` was called.

File: services/horizon/internal/render/sse/stream_test.go [hunk 3]
```diff
@@ -1,24 +1,126 @@
...
 // Tests that heartbeat events are sent by Stream.
-func TestStream_SendHeartbeats(t *testing.T) {
-	ctx, _ := test.ContextWithLogBuffer()
-	w := httptest.NewRecorder()
-	stream := NewStream(ctx, w).(*stream)
+func (suite *StreamTestSuite) TestStream_SendHeartbeats() {
 	// Set heartbeat interval to a low value for testing.
-	stream.interval = 500 * time.Millisecond
-	stream.Init()
+	suite.stream.(*stream).interval = 500 * time.Millisecond
```
Comment (bartekn) [inline services/horizon/internal/render/sse/stream_test.go]: I know that I accepted the previous PR with this test but it's checking if heartbeat was sent once. We should probably change it so something like:
```go
suite.stream.(*stream).interval = 100 * time.Millisecond // Send heartbeats faster
suite.stream.Init()
time.Sleep(120*time.Millisecond)
// Check if # of occurences of "\n:heartbeat" = 1
time.Sleep(120*time.Millisecond)
// Check if # of occurences of "\n:heartbeat" = 2
...
```

Comment (lechengfan) [reply services/horizon/internal/render/sse/stream_test.go]: That sounds good, thanks.

**Pull request merged**